### PR TITLE
Added update experience functionality

### PR DESCRIPTION
**Pull request by andyjianzhou**
- [x] Solves #7 
- Created unittest to verify editing existing experience based on ID

### CODE REVIEW

**Comment by andyjianzhou** [comment]
@Pradyuman7 
Ready for review

File: app.py [hunk 1]
```diff
@@ -56,6 +56,24 @@ def experience():
...
     
+    if request.method == "PUT":
+        index = request.args.get("index")
+        if index is None:
+            return jsonify({"error": 'Index not provided'}), 400
+        if not index.isnumeric():
+            return jsonify({"error": "Index must be a number"}), 400
+        
+        index = int(index)
+        if not (0 < index <= len(data["experience"])):
```
Comment (Pradyuman7) [inline app.py:67]: I think there's a small out of range issue here, might use < intead of <=

Comment (andyjianzhou) [reply app.py:67]: No, that is correct, it should be <= because the ids are 1 indexed. If you remove it, and try it on pytest, it will prompt an error

**Review by Pradyuman7 — APPROVED** [review]
LGTM 🥇

**Pull request merged**